src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -504,5 +513,33 @@ private static string[] SplitFlagsEnum(string value)
 #endif
                 );
         }
+
+        private static bool ContainsSpecialChar()
+        {
+#if NETCOREAPP
+            string[] names = Enum.GetNames<T>();
+#else
+            string[] names = Enum.GetNames(typeof(T));
+#endif
+
+            foreach (string name in names)
+            {
+                foreach (char cha in name)
+                {
+#if NET7_0_OR_GREATER
+                    if (!char.IsAsciiLetterOrDigit(cha) && cha != '_')
```

Comment:
This seems too restrictive, for example it would disqualify non-ascii letters which are generally supported per [identifier naming rules](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-rules). 

I would recommend you add a more conservative check, e.g. one that only detects comma or whitespace characters.

Reply:
make sense.